### PR TITLE
feat: added web development free courses for beginners

### PR DESCRIPTION
**Pull request by Jeyaprabakar01**
## What does this PR do?
Add resource(s)

### Description
These resources are for those who want to get started with Front-end web development. It covers such things as Git,GitHub, HTML, CSS, JavaScript, and React.

### Why is this valuable (or not)?
The provided resources are easy to follow and fully free. They can learn by following step-by-step instructions. so as beginners, they can easily learn these skills.

### How do we know it's really free?
The provided courses are from YouTube and Microsoft Learn. Always, they both are fully free.

### For book lists, is it a book? For course lists, is it a course? etc.
It's a course.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by eshellman** [comment]
For microsoft Learn, I think we'll want to list the paths and not the modules.

**Comment by Jeyaprabakar01** [comment]
> For microsoft Learn, I think we'll want to list the paths and not the modules.

yeah that's correct. Now I modified the courses with the paths

**Comment by eshellman** [comment]
For youtube, we prefer "playlist" links and the name of the producer/creator. docs/CONTRIBUTING.md#platforms-and-access-notes

**Comment by Jeyaprabakar01** [comment]
> I modified the courses with the paths

 I modified the youtube videos with the playlists and pushed

**Comment by Jeyaprabakar01** [comment]
> For youtube, we prefer "playlist" links and the name of the producer/creator. docs/CONTRIBUTING.md#platforms-and-access-notes

I modified the youtube videos with the playlists and pushed

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -649,6 +649,7 @@
 
 ### Git
 
+* [A Collaborate with others with Markdown and GitHub Pages](https://learn.microsoft.com/en-us/training/paths/collaborate-markdown-github-pages/) - Microsoft Learn
```
Comment (eshellman) [inline courses/free-courses-en.md:652]:
```suggestion
* [Collaborate with others with Markdown and GitHub Pages](https://learn.microsoft.com/en-us/training/paths/collaborate-markdown-github-pages/) - Microsoft Learn
```

File: courses/free-courses-en.md [hunk 2]
```diff
@@ -816,6 +819,7 @@
 
 ### JavaScript
 
+* [30 JavaScript Projects For Beginners](https://youtube.com/playlist?list=PLjwm_8O3suyOgDS_Z8AWbbq3zpCmR-WE9&si=fofo8k2c5sjQPOKq) - GreatStack
```
Comment (eshellman) [inline courses/free-courses-en.md:822]:
```suggestion
* [30 Days, 30 JavaScript Projects For Beginners to Practice](https://youtube.com/playlist?list=PLjwm_8O3suyOgDS_Z8AWbbq3zpCmR-WE9&si=fofo8k2c5sjQPOKq) - GreatStack
```

File: courses/free-courses-en.md [hunk 3]
```diff
@@ -922,6 +928,7 @@
 * [React](https://progate.com/languages/react) (progate) *(account required)*
 * [React basic in just 1 hour](https://www.udemy.com/course/react-basic-in-just-1-hour/) (Udemy)
 * [React Context & Hooks Tutorial](https://www.youtube.com/playlist?list=PL4cUxeGkcC9hNokByJilPg5g9m2APUePI) - The Net Ninja
+* [React Course For Beginners](https://www.youtube.com/playlist?list=PLpPqplz6dKxW5ZfERUPoYTtNUNvrEebAR) - PedroTech
```
Comment (eshellman) [inline courses/free-courses-en.md:931]:
```suggestion
* [ReactJS Course For Beginners 2023](https://www.youtube.com/playlist?list=PLpPqplz6dKxW5ZfERUPoYTtNUNvrEebAR) - PedroTech
```

File: courses/free-courses-en.md [hunk 4]
```diff
@@ -833,6 +837,7 @@
 * [Javascript Fundamentals](https://www.udemy.com/course/javascriptfundamentals) - Bharath Thippireddy (Udemy)
 * [JavaScript Leetcode](https://www.youtube.com/playlist?list=PLRKOqqzwh75hcG_D4xjUgbg_BAF3MLtoh) - Endeavour Monk
 * [JavaScript Mini Course 2020](https://www.udemy.com/course/javascript-essentials-mini-course/) - Kalob Taulien (Udemy)
+* [JavaScript tutorial for beginners](https://youtube.com/playlist?list=PL9bD98LkBR7P16BndaNtP4x6Wf5Ib85Hm&si=u_pP2hpNXqa2_2QY) - Tech2 etc
```
Comment (eshellman) [inline courses/free-courses-en.md:840]: ****
```suggestion
* [JavaScript Projects For Beginners \| Easy To Advance](https://youtube.com/playlist?list=PL9bD98LkBR7P16BndaNtP4x6Wf5Ib85Hm) - Tech2 etc
```

**Review by eshellman — CHANGES_REQUESTED** [review]
Don't invent titles! Use the title given by the resource.
Remove tracking ids from urls
when you've made changes, please add `(changes made)` to the title of your PR

**Pull request merged**